---
title: "Negligible heritability of language laterality assessed by functional transcranial
  Doppler ultrasound: a twin study"
author: "DVM Bishop"
date: "16/05/2019"
output:
  word_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
#---------------------------------------------------------------
knitr::opts_chunk$set(echo = TRUE,message = FALSE,warning=FALSE)
batesinfo <- 1 #bits for tim bates
#to do
# AE model with ordinal - is this same as just using ranks? - NO
# Alternative test of power using simulated data
# ? More analysis of original Medland handedness twin data
require(yarrr)
require(stargazer) #Hlavac, Marek (2018). stargazer: Well-Formatted Regression and Summary Statistics Tables.
# R package version 5.2.1. https://CRAN.R-project.org/package=stargazer 
require(tidyverse)
require(OpenMx)
require(MASS)
require(psych) #for phi coeff
require(ggpubr)

source('http://ibg.colorado.edu/cdrom2016/verhulst/Power/powerFun.R')
#see http://www.people.vcu.edu/~bverhulst/power/powerScript.R
#mxOption(NULL, 'Default optimizer', 'NPSOL') #?? this gets 'could not find function mxOption'
```
```{r readdata, echo=FALSE}
#---------------------------------------------------------------
#Read data before any text as need to refer to Ns etc computed here.
#Version from April 2019 includes LI indices using means rather than peaks, and also has measure of
#cross-trial consistency
#---------------------------------------------------------------
mydir<-"/Users/dorothybishop/Dropbox/ERCadvanced/project twin kids/Project_files/"
#mydir<-"c:/Users/pthompson/Dropbox/project twin kids/Project_files/"
myfile<-"TwinsData_DATA_2019-04-12_1525.csv"
mydata<-read_csv(paste0(mydir,"Data/",myfile))
#str(mydata)
w<-which(colnames(mydata)=='include')  #this relates to exclusion criteria re IQ,autism etc : not used here
colnames(mydata)[w]<-'myinclude' #to avoid problems with having 'include' as column name
```

```{r makeLIcat}
mydata$lat_index<-as.numeric(mydata$myli_mean)  #the original laterality_index value in file is LI from peak
#my_li_mean was added in April 2019
mydata$myse_mean<-as.numeric(mydata$myse_mean)
mydata$my_li_mean_even<-as.numeric(mydata$my_li_mean_even)
mydata$my_li_mean_odd<-as.numeric(mydata$my_li_mean_odd)

mydata$qhp_handedness<-mydata$qhp_freq_r+.5*mydata$qhp_freq_e

#categorise laterality as 1=L, 2=bi, 3 = R depending on whether mean value is outside 1.96*SE limit
#NB can use 1.96 or 1.65 here, but 1.96 is same as used by Wilson and Bishop
CIterm<-1.96
mydata$li_cat<-NA
w<-which(mydata$lat_index>0)
mydata$li_cat[w]<-1 #default is L laterlised for those with +ve values
x<-mydata$lat_index[w]-CIterm*mydata$myse_mean[w]
#need to be careful here re indices - we are now working with the w subset and finding which are -ve
y<-which(x<0)
w1<-w[y] #these should be the indices from the original full file with L sided lat but large SE -> bilateral
mydata$li_cat[w1]<-2

#Now do the same for R lateralised
w<-which(mydata$lat_index<0)
mydata$li_cat[w]<-3 #default is R lateralised for those with -ve values
x<-mydata$lat_index[w]+CIterm*mydata$myse_mean[w] #this time we add the 1.96*SE term
#need to be careful here re indices - we are now working with the w subset and finding which are -ve
y<-which(x>0)
w1<-w[y] #these should be the indices from the original full file with R sided lat but large SE -> bilateral
mydata$li_cat[w1]<-2

#temp<-dplyr::select(mydata,lat_index,myse_mean,li_cat) #view temp to check output
temp<-filter(mydata,n_trials>11) #same as Wilson and bishop, have at least 12 trials
table(temp$li_cat,temp$lateralised_category)
#Mean method gives more bilateral cases - I guess not surprising. 

#check correlation between old and new LI methods
oldnewLIcor<-cor(temp$laterality_index,temp$lat_index,use='pairwise.complete.obs')
oldnewLIcordf<-length(temp)-2

#check split half reliablity
splithalfr<-cor(temp$my_li_mean_even,temp$my_li_mean_odd,use='pairwise.complete.obs')
```
```{r multitasklat}
#-------------------------------------------------------
# Create a variable that has categorical laterality for 2 handedness tasks and LI
#-------------------------------------------------------
mydata$lat3<-2 #default is mixed laterality = 2
w<-which(mydata$qhp_handedness<11&mydata$ehp_handedness<6&mydata$li_cat>1)
mydata$lat3[w]<-3 #reversed laterality - all either bilat or R
w<-which(mydata$qhp_handedness>10.5&mydata$ehp_handedness>5.5&mydata$li_cat==1)
mydata$lat3[w]<-1 #typical laterality

#also just do a count of N tasks in conventional direction - taskssum
mydata$taskssum<-0
w<-which(mydata$qhp_handedness>10)
mydata$taskssum[w]<-mydata$taskssum[w]+1 
w<-which(mydata$ehp_handedness>5)
mydata$taskssum[w]<-mydata$taskssum[w]+1 
w<-which(mydata$li_cat==1)
mydata$taskssum[w]<-mydata$taskssum[w]+1 

w<-which(is.na(mydata$li_cat))
mydata$lat3[w]<-NA
mydata$taskssum[w]<-NA

```

```{r selectcols, echo=FALSE}
#----------------------------------------------------------
# Produce data frame of selected columns 
#----------------------------------------------------------
data.short<-filter(mydata,zygosity<9)
data.short<-dplyr::select(mydata,record_id,fam_id,age_at_test,female,zygosity,n_language_low,myinclude,twin,n_trials, lat_index,qhp_handedness,ehp_handedness,lang_probs,li_cat,lat3,ehp_handedness,qhp_handedness,taskssum)
#identify cases to exclude because of less than 12 trials on doppler
w<-c(which(data.short$n_trials<12),which(is.na(data.short$n_trials)))
data.short$doppexcl <- 0
data.short$doppexcl[w] <-1

```

```{r doublentry, echo=FALSE}
#-------------------------------------------------------
# Create double entry file with twin 1 and 2 aligned
# Assumes twins from pair are in adjacent rows
#-------------------------------------------------------
nrec<-nrow(data.short)
nvar<-ncol(data.short)
nuorder2<-c(seq(from=2,to=nrec,by=2),seq(from=1,to=nrec,by=2)) #odd nums
nuorder1<-c(seq(from=1,to=nrec,by=2),seq(from=2,to=nrec,by=2)) #even nums
doubledata<-cbind(data.short[nuorder1,],data.short[nuorder2,])
range2<-(nvar+1):(nvar*2)
colnames(doubledata)[range2]<-paste0(colnames(doubledata)[1:nvar],2)

#recode all DZ as 2, MZ as one
doubledata$MZDZ<-2
w<-which(doubledata$zygosity==1)
doubledata$MZDZ[w]<-1

#add a column code to identify whether gender  MM or MF or FF
doubledata$gender2 <-as.factor(1+doubledata$female+doubledata$female2)
#check all aligned
check<-sum(doubledata$fam_id-doubledata$fam_id2)
if(check>0) {print('Twins not aligned!!!')}

doubledata$langpair <- paste0(doubledata$lang_probs,doubledata$lang_probs2)
mylangtab <- table(doubledata$langpair[1:(nrow(doubledata)/2)])
doubledata$langconcord <-2
w <-which(doubledata$lang_probs==0)
doubledata$langconcord[w] <-1
w <-which(doubledata$lang_probs2==0)
doubledata$langconcord[w] <-1
w1<- which(doubledata$langpair=='00')
doubledata$langconcord[w1] <-0
conc.table<-table(doubledata$MZDZ,doubledata$langconcord,doubledata$twin)
par(mfrow=c(1,2))
par(mar=c(0,0,1,0))
slices <- conc.table[1,2:3,1]
#lbls <- c("One twin","Both twins")
lbls <-c('','')
pie(slices, labels=lbls, main="Language concordance: \nIdentical twins")
slices <- conc.table[2,2:3,1]

pie(slices, labels =lbls, main="Language concordance: \nNonidentical twins")

```

```{r exclusions, echo=FALSE}
#-------------------------------------------------------
# Exclude those with unusable data - either twin
#-------------------------------------------------------


doubledata$doppout <-doubledata$doppexcl+doubledata$doppexcl2
w<-which(doubledata$doppout==2)
doubledata$doppout[w]<-1

#remove case of extreme LI, defined as +/-10
w<-c(which(abs(doubledata$lat_index)>12),which(abs(doubledata$lat_index2)>12))
doubledata$doppout[w]<-1 #picks up case with LI of -21,which is more than 5 SD from mean!
#There seems no good reason for this extreme score- data look clean! But so extreme best removed.

doubledata2<-filter(doubledata,doppout<1)

#unit of analysis is twin pair excluded, so need to divide by 2 as one row per twin

Npairwithhand <- nrow(doubledata)/2
Npairwithdopp<-nrow(doubledata2)/2
```

##Abstract

## Introduction
Lateralisation of language and motor function in humans are traits with two notable features: First, there is a pronunced population bias to one side - to the left for lateralisation of language in the brain, and to the right for handedness; Second there is individual variation in the extent of bias; a minority of individuals show a reversal of the typical pattern, and others show little or no asymmetry.

Other primates show some indications of lateralised functions, but humans are quite distinctive in their strong population bias to right-handedness. It is often assumed that evolution of manual laterality is related to development of a complex and lateralised language faculty in humans, but the biological origins of the population bias are not well understood for either trait. 

Twins provide a useful natural experiment for estimating the contribution of genetic variation to individual differences in a trait. The twin method uses comparisons of similarity of identical (monozygotic or MZ) twins versus non-identical (dizygotic or DZ twins) to derive estimates of the relative contributions of genetic variants, environment shared by the twins, and other twin-specific influences (including chance) on individual variation in a trait. This is best understood intuitively by imagining a variety of hypothetical situations. In the first, the trait is solely determined by random chance: in that case, two members of a twin pair (A and B) would be no more similar than two unrelated people, and in a sample of twins, the correlation between twin A and twin B would be zero, regardless of zygosity. In a second fictitious scenario, the trait is solely determined by an environmental factor common to both twins, such as home environment. In that case, there would be perfect correlation between the traits for twin A and twin B. In the third scenario, the trait is determined solely by genes: because MZ twins are genetically identical, the correlation between two members of a twin pair will be 1, but for DZ twins, who on average have 50% of their segregating genes in common, the correlation will be .5. 

In practice, our goal is not to attribute variation in the observed trait (phenotype) to one cause or the other, but rather to assume their effects are additive and estimate their relative contribution. We specify that the total variance in a trait, v is equal to a^2^ + c^2^ + e^2^, where a is additive genetic variance, c is shared (common) environment and e is random, nonshared environment. Similarity between pairs of MZ twins is the sum of genetic and shared environmental influences, a^2^ + c^2^, and similarity between pairs of DZ twins is .5 a^2^ + c^2^, so we can estimate heritability (a^2^) as twice the difference in correlation between MZ and DZ twins, and then obtain the values for c^2^ and e^2^ by simple algebra (Sham, 1998).

Twin studies of manual laterality have shown that, contrary to conventional wisdom, individual differences in handedness are largely determined by chance, with genetic variation playing a relatively minor role. A meta-analysis of 35 twin studies estimated heritability of around .23, with shared environment effect of zero (Medland, 2006). Studies of language laterality are far less numerous, because of the difficulty and expense of studying large numbers of twins. Table 1 summarises results from studies that have looked at structural and functional asymmetries in twins, and one can see that the estimates of heritability are again low, and in keeping with those previously reported for handedness.

To do: Table 1 - summary of studies of structural/functional asymmetry

As far as we are aware, to date there has not been a twin study that uses a direct, functional measure of language lateralisation. We report genetic analysis from a study of `r Npairwithdopp` twin pairs, showing that, consistent with previous studies of handedness and brain asymmetry, chance (or environmental factors not shared between twins) plays the major role in determining individual differences.

Our sample consists of twin children recruited for a study of the genetic bases of developmental language disorder (DLD), for whom language lateralisation was assessed using functional transcranial Doppler ultrasound (fTCD). Data from these children have previously been reported in the context of an analysis focusing on relationships between cerebral lateralisation and language functioning (Wilson & Bishop, 2018). That analysis found no difference in language laterality between children with language disorders and those with typical language development, although the reliability of the laterality index obtained with this measure was good (split-half reliability for odd and even trials = .84). Furthermore, comparison with a previous study using the same methods confirmed that language lateralisation in twins did not differ from that observed in single-born children. This sample provides a useful opportunity to fill a gap in the literature with an analysis comparing MZ and DZ twins in order to estimate the relative contribution of genetic and non-genetic variation to individual differences in language laterality.  

##Methods
###Participants
For a detailed account of selection of participants, see Wilson and Bishop (2018). In brief, we recruited `r Npairwithhand ` pairs of twins aged 6 years 0 months to 11 years 11 months, using a sampling approach with the aim of including around 75% twin pairs where one or both had language or literacy difficulties. Handedness assessments were completed for all children, and language laterality assessment (see below) was available for `r Npairwithdopp` pairs. The breakdown of the sample by zygosity and gender is shown in Table 2.

```{r table2, echo=FALSE}
#--------------------------------------------------------------
# Create table N twin pairs by zygo and sex
#--------------------------------------------------------------
doubledata$zygosex<-10*doubledata$zygosity+doubledata$female
w<-which(doubledata$zygosex==31)
doubledata$zygosex[w] <-30
doubledata$zygosex<-as.factor(doubledata$zygosex)
levels(doubledata$zygosex) <- c('MZ female','MZ male','DZ female','DZ male','DZ male/female')
tab2 <- table(doubledata$zygosex,doubledata$doppout)
tab2f<-data.frame(cbind(tab2[1:5],tab2[1:5],tab2[6:10]))
tab2f[,1]<-tab2f[,2]+tab2f[,3]
tab2f[,1:2]<-tab2f[,1:2]/2
tab2f<-tab2f[,1:2]
row.names(tab2f)<-levels(doubledata$zygosex)
colnames(tab2f)<-c('All','With fTCD')
stargazer(tab2f,type='text',title='Table 2: N twin pairs by zygosity and sex',summary=FALSE)
#--------------------------------------------------------------

```
###Zygosity determination
DNA was available for 191 twin pairs who were compared across 250,875 SNPs. All gave unambiguous zygosity signals on Identity by State (IBS), i.e., the proportion of SNPs for which any given twin pair share genotypes: this was either close to 1.0 (MZ) or close to 0.5 (DZ). For twins with missing or inadequate DNA samples we relied on parental report of zygosity.

###Laterality assessment
1. Handedness

Hand preference was assessed using a hand preference battery based on items from the Edinburgh Handedness Inventory (EHI) (Oldfield, 1971), modified to exclude one item deemed unsuitable for children (striking a match). Children were asked to demonstrate each of ten actions: writing, drawing, throwing a ball, using scissors, using toothbrush, cutting with a knife, using a spoon, using a broom (upper hand), taking the lid off a box, and dealing cards. One point was awarded for exclusive right hand use, zero points for left hand use, and half a point if both hands were used, giving a score ranging from zero to ten.

Strength of hand preference was assessed using the Quantification of Hand Preference (QHP) task (Bishop et al., 1996), which measures the tendency to continue to use the preferred hand when cards are picked up from different spatial locations. Three cards are set out in each of seven positions extending at 30 degree intervals from the left to the right of the child’s midline. The child is not told that handedness is being assessed, and treats the task as a picture-name matching game, where they have to pick up the named card and put it in a centrally-placed box. The same quasi-random order of positions was used for all children, starting with a card at the midline and continuing until the child had picked up and placed  three cards at each of seven locations, to give a total of 21 trials. For each card, two points were awarded for right-handed use, zero points for left-handed use, and one point if the card was transferred from one hand to another in the course of placing it in the box.

2. Language laterality

Language laterality was assessed using functional transcranial Doppler ultrasound (fTCD) while the child described short episodes from a story presented as an animation. A video demonstration of this procedure is available from Bishop et al (2010). Transcranial Doppler ultrasound is used in medical contexts to assess the integrity of the cerebral blood vessels.  For assessing cerebral lateralisation, left and right ultrasound probes are attached to a headset and positioned so as to detect lateralised changes in blood flow in the middle cerebral arteries. 

On each trial, the child silently views a 12 s clip from a cartoon that included sounds but no speech. A response cue appears when the video clip finishes to indicate the start of a 10 s period during which the child is asked to describe what happened in the cartoon. A second cue then indicates that the child should stop talking and relax. This paradigm has previously been found to have good validity and reliability (Bishop et al. 2009).

A maximum of 30 trials was administered, depending on the child's tolerance of the procedure. The child's verbal responses were recorded and subsequently transcribed, and the examiner noted behaviour during the procedure.  Trials were excluded if the child either spoke during a silent period, or failed to talk during the 'talk' period: these need to be omitted because they invalidate the trial, which involves comparing the period when the child talks with a baseline period when no talking occurs. 

The analysis of the animation task data consists of a standard sequence of processing steps, following original work by Deppe et al (1997). We used a custom script written in R (R core team, 2013) for data processing. This included an initial step of identifying trials where there was very brief signal dropout (affecting one datapoint) and interpolate the mean value in such cases. Trials with more prolonged signal dropout were discarded. After these preliminary steps, heart cycle integration was applied to remove the heartbeat, followed by signal normalisation, artefact rejection, epoching and baseline correction.The averaged left and right velocity plots were subtracted to give a difference waveform.

In our previous report of data from this sample, we used the conventional approach for obtaining a laterality index based on the peak difference (maximum or minimum) in a period of interest, predefined as 4 to 14 seconds after the cue to speak. Our subsequent studies suggested that this method is not ideal, because there are cases where the difference wave shifts from positive to negative, or vice versa, within the period of interest, and quite minor difference in size of positive and negative peaks can determine whether laterality is coded as left or right.  Accordingly, in more recent studies, we have calculated a laterality index (LI) as the mean amplitude of blood flow velocity difference in the whole period of interest. The correlation between this mean-based LI and the traditional peak-based LI is very strong: Pearson correlation = `r round(oldnewLIcor,3)`, DF = `r oldnewLIcorDF`, but the distributions differ. The split-half reliability, based on correlation if LIs from odd and even trials is closely similar for the two methods, r = `r round(splithalfr,2)`. The conventional method, however, gives a non-normal distribution of laterality indices, with a point of rarity around zero, which appears to be a spurious artefact of the method of computation.

Following Groen et al (2012) we excluded data from children who had fewer than 12 accepted trials, as the LI is likely to be unreliable when based on such a small amount of data. In addition, data were excluded for one twin pair where one child's laterality index was more than 5 SD from the mean. 

The standard error of the LI for an individual was computed from the LI obtained across individual trials. This allowed us to consider whether the LI was significantly different from zero, i.e. whether the 95% confidence interval spanned zero. Where this was the case, laterality was categorised as left or right, and where the LI was not significantly different from zero, the laterality was coded as bilateral. Note that coding of bilateral laterality can result if data are merely noisy.

3. Consistency of manual/language laterality.

Prominent genetic theories by Annett (1985) and McManus (1985) have proposed that observed distributions of handedness are the result of a mixture of two underlying genotypes: one that shows a population bias and the other that has handedness randomly determined. If the typical bias affected both manual and language laterality, this should lead to a typical pattern of right-handedness and left-brainedness for language; the greater the departure from this pattern, the greater the likelihood that the individual has the atypical, unbiased genotype. To evaluate this idea, individuals were categorised as typical or atypical on each measure, and an overall measure derived consisting of the number of tasks with typical laterality (R for handedness and L for language). For the language measure, left-lateralisation was coded when the LI was positive and the 95% confidence interval around the LI did not include zero. For the handedness measure, R-handedness was coded when more actions/reaches were performed with the right hand than with the left.

## Procedure  
(This section is copied from Wilson & Bishop, 2018). Ethical approval was obtained for the study in 2011 from the Berkshire NHS Research Ethics Committee (reference 11/SC/0096), and data collection started in August of that year, finishing in October 2016. Where families had expressed interest in the study, they were interviewed by telephone to assess whether the children were likely to meet inclusion criteria, and an appointment was made to see the twins at home or at school, depending on parental preference. Families were widely dispersed around the UK, including Northern Ireland, Scotland, Wales and England, so testing was scheduled where possible to minimise travel. During the course of recruitment, which lasted for a period of five years, a total of eight research assistants as well as the senior author were involved in assessing children. In some cases, two testers worked together, each seeing one twin, and in others a single tester saw both children sequentially. The assessment was conducted in a single session lasting between 2-3 hours per child, with breaks where needed.

```{r powertab, echo=FALSE}
#------------------------------------------------------------------------
# DVMB version of powerFun, with just AE in model
# Used to see how low a value of a2 is detected in AE model with this sample size
#------------------------------------------------------------------------
powertable <-data.frame(matrix(NA,nrow=20,ncol=5)) #initialise table to hold results
myaa<-vector() #blank vector: will hold smallest value of aa with lower CI > 0
colnames(powertable)<-c('MZ/DZ Npairs','true.aa','est.aa','se.est','lowerCI')
thisrow<-0
for (myn in 1:2){ #estimate for both sample sizes
for (myadd in seq(5,50, by=5)){ #estimate for a2 values from .05 to .5 in steps of .5
  thisrow<-thisrow+1
aa=myadd/100 # 
Nmz=tab2f[1,myn]+tab2f[2,myn] #take Ns from previous tab2f
Ndz=tab2f[3,myn]+tab2f[4,myn]+tab2f[5,myn]


nv <- 1 #N variables
ntv <- nv*2 #N columns (twins)

ee <- sqrt(1 - aa^2)
AA <- aa^2
EE <- ee^2

mzMat <- matrix(c(AA + EE, AA, AA, AA + EE),2)
dzMat <- matrix(c(AA + EE, .5*AA, .5*AA, AA + EE),2)

mzData <- mvrnorm(Nmz , mu = c(0,0), mzMat, empirical = T)
dzData <- mvrnorm(Ndz , mu = c(0,0), dzMat, empirical = T)

selVars <- paste("t", 1:2, sep = "")
colnames(mzData) <- colnames(dzData) <- selVars
MZdata  <-  mxData(mzData, type="raw" )
DZdata  <-  mxData(dzData, type="raw" )
#Used these 2 lines to look at impact of log transform
#It wipes out heritability!
#MZdata  <-  mxData(log(mzData+5), type="raw" )
#DZdata  <-  mxData(log(dzData+5), type="raw" )
#Used these 2 lines to look at impact of exp transform
#It makes heritability estimates v unstable, but often huge!
#Presumably because of large effect of extreme cases
# MZdata  <-  mxData(exp(mzData), type="raw" )
# DZdata  <-  mxData(exp(dzData), type="raw" )



a <-  mxMatrix( type="Lower", nrow=nv, ncol=nv, free=TRUE, values=.6, label="a11", name="a" )
e <-  mxMatrix( type="Lower", nrow=nv, ncol=nv, free=TRUE, values=.6, label="e11", name="e" )

A <-  mxAlgebra( expression=a %*% t(a), name="A" )
E <-  mxAlgebra( expression=e %*% t(e), name="E" )

Mean    <-  mxMatrix( type="Full", nrow=1, ncol=nv, free=TRUE, values= 0, label="mean", name="Mean" )
expMean <-  mxAlgebra( expression= cbind(Mean,Mean), name="expMean")

expCovMZ <- mxAlgebra( expression= rbind  ( cbind(A+E , A),cbind(A   , A+E)), name="expCovMZ" )
expCovDZ <- mxAlgebra( expression= rbind  ( cbind(A+E     , 0.5%x%A),cbind(0.5%x%A , A+E)),  name="expCovDZ" ) 

obs <-  list(a,e,A,E,Mean,expMean,expCovMZ,expCovDZ)

fun <- mxFitFunctionML()
mzExp <- mxExpectationNormal(covariance="expCovMZ", means="expMean", dimnames=selVars )
dzExp <- mxExpectationNormal(covariance="expCovDZ", means="expMean", dimnames=selVars )

MZ <- mxModel("MZ", obs, MZdata, fun, mzExp)
DZ <- mxModel("DZ", obs, DZdata, fun, dzExp)

aeFun <- mxFitFunctionMultigroup(c("MZ","DZ"))
ae <- mxModel("AE", MZ, DZ, fun, aeFun)

aeFit <- mxRun(ae, silent = T)
summary(aeFit)

pars <- round(cbind(aeFit$output$estimate, aeFit$output$standardErrors), 3)
colnames(pars) <- c("Estimates", "Std. Err.")

powertable[thisrow,1]<-paste0(Nmz,'/',Ndz)
powertable[thisrow,2]<-aa
powertable[thisrow,3:4]<-pars[1,1:2]
powertable[thisrow,5]<-round(pars[1,1]-1.96*pars[1,2],3)
#Find the value of a2 where lower CI is just above 0
if (powertable[thisrow,5]<0){thisaa<-thisrow}
}
  myaa<-c(myaa,powertable[(thisaa+1),2])
}
#myaa holds 2 values: one for larger and one for smaller sample size
#These correspond to smallest values of aa detectable as different from zero with this sample size

```
## Data analysis
1. Power analysis

As noted above, the usual approach to twin analysis is ACE decomposition, typically implemented using structural equation modeling with maximum likelihood estimates (Rijsdijk & Sham, 2002). Large samples are needed to accurately estimate additive genetic (a^2^), shared environmental (c^2^) influences, both of which lead to positive covariance between two members of both MZ and DZ pairs: they are distinguished by the fact that genetic influence leads to greater covariance for MZ and DZ twins. In the context of laterality, however, prior studies have found shared environmental influences to be negligible in adequately powered twin studies, and it is safe to ignore the c^2^ term. This simplifies the analysis, making it possible to detect genetic effects with smaller samples. A power analysis using a modified version of the powerFun function in OpenMx showed that for the larger sample of twins for whom handedness data were available, there is power to detect heritability, a^2^, of `r round(myaa[1]^2,2)` or greater, and for the smaller sample with language laterality data, an effect of `r round(myaa[2]^2,2)` or greater is detectable.

2. AE modeling



## Results
Figure 1 shows scatterplots depicting the association of the laterality indices between two members of a twin pair; there is no indication of any association between the LI indices for two members of a twin pair.  

```{r computespearman,echo=FALSE}
makespearman <- function(thismz,thisdz){
  #create text with Spearman correlation values.
#NB need to compute these from *unjittered* data 
MZrs <- cor(thismz$thisx,thismz$thisy, use="pairwise.complete.obs",method='spearman')
MZn<-nrow(filter(thismz,!is.na(thisx),!is.na(thisy)))/2 #divide by 2 for N pairs
DZrs <- cor(thisdz$thisx,thisdz$thisy, use="pairwise.complete.obs",method='spearman')
DZn<-nrow(filter(thisdz,!is.na(thisx),!is.na(thisy)))/2 #divide by 2 for N pairs
mylabelbit<-c(paste0('MZ: r (',(MZn-2),') = ',round(MZrs,3)),paste0('DZ: r (',(DZn-2),') = ',round(DZrs,3)))
for (i in 1:2){
while(nchar(mylabelbit[i])<18){
  mylabelbit[i]<-paste0(mylabelbit[i],'0')
}
}
return(mylabelbit)
}
```

```{r scatterfunction,echo=FALSE}
makescatter <- function(mytitle,xpos,ypos,filetitle,vline,mylabeldf,mylabelbit){
  g <-ggscatter(doubledata[1:singlerows,], x = 'myx', y = 'myy',
            xlab='Twin 1',ylab='Twin 2',
            shape = "Zygosity",
            # Change point shape by groups 
            color = "Zygosity", palette = get_palette("aaas", 2),
            # Color by groups 
            title = mytitle  )+
  #  stat_cor(aes(color = Zygosity), label.x =xpos,
  #           label.y=ypos,method='spearman',show.legend=FALSE) +
  
    geom_vline(xintercept=vline,linetype="dashed")+
    geom_hline(yintercept = vline,linetype="dashed") +
   annotate("text",label=mylabelbit[1], x = xpos[1],y = ypos[1],size=3)+
   annotate("text",label=mylabelbit[2], x = xpos[2],y = ypos[2],size=3)

  ggsave(filetitle,width=4,height=4)
  return(g)
}

```

```{r scatterplot,echo=FALSE}
#------------------------------------------------------------------------
#scatterplot and correlation by zygosity as MZ/DZ: do for LI and handedness measures
#------------------------------------------------------------------------
#Use the makescatter function defined above for all the plots

doubledata$Zygosity<-as.factor(doubledata$MZDZ) #Initial caps for version that is factor with 2 values
levels(doubledata$Zygosity)<-c('MZ','DZ')
singlerows<-nrow(doubledata)/2

#EHI plot
#compute spearman correlation using function and assemble into string
doubledata$thisx<-doubledata$ehp_handedness
doubledata$thisy<-doubledata$ehp_handedness2
thismz<-filter(doubledata,MZDZ==1)
thisdz<-filter(doubledata,MZDZ==2)
mylabelbit<-makespearman(thismz,thisdz)

#add jitter to points before plotting
myjitter<-rnorm(nrow(doubledata))/4.5 #need to add jitter to points for plotting
doubledata$myx<-doubledata$ehp_handedness+myjitter
myjitter<-rnorm(nrow(doubledata))/4.5
doubledata$myy<-doubledata$ehp_handedness2+myjitter
mytitle<-'Edinburgh Handedness Inventory'
filetitle<-'ehiplot.png'
vline=5.5
xpos <- c(2.2,2.2)
ypos <- c(4.5,3.5)
g1<-makescatter(mytitle,xpos,ypos,filetitle,vline,mylabeldf,mylabelbit)
```

###########################################################################################
```{r QHPplot}

#compute spearman correlation using function and assemble into string
doubledata$thisx<-doubledata$qhp_handedness
doubledata$thisy<-doubledata$qhp_handedness2
thismz<-filter(doubledata,MZDZ==1)
thisdz<-filter(doubledata,MZDZ==2)
mylabelbit<-makespearman(thismz,thisdz)

myjitter<-rnorm(nrow(doubledata))/4
doubledata$myx<-doubledata$qhp_handedness+myjitter
myjitter<-rnorm(nrow(doubledata))/4
doubledata$myy<-doubledata$qhp_handedness2+myjitter
xpos<-c(3,3) #coordinates for correlation on plot
ypos<-c(8,6)
mytitle<-'Quantification of Hand Preference'
filetitle<-'qhpplot.png'
vline=11
g2 <-makescatter(mytitle,xpos,ypos,filetitle,vline,mylabeldf,mylabelbit)
```
###########################################################################################
```{r LIplot}
w<-which(doubledata$doppout==1)
doubledata$lat_index[w]<-NA #this removes outlier value 
doubledata$lat_index2[w]<-NA

#compute spearman correlation using function and assemble into string
doubledata$thisx<-doubledata$lat_index
doubledata$thisy<-doubledata$lat_index2
thismz<-filter(doubledata,MZDZ==1)
thisdz<-filter(doubledata,MZDZ==2)
mylabelbit<-makespearman(thismz,thisdz)
doubledata$myx<-doubledata$lat_index #no need for jitter with LI
doubledata$myy<-doubledata$lat_index2
mytitle<-'Laterality Index (fTCD)'
xpos<-c(4,4) #coordinates for correlation on plot
ypos<-c(10,9)
filetitle<-'LIplot.png'
vline=0
g3<-makescatter(mytitle,xpos,ypos,filetitle,vline,mylabeldf,mylabelbit)


###########################################################################################

```

```{r combined plots}

ggarrange(g1,g2,g3,ncol=1,nrow=3,common.legend = TRUE,heights=c(1,1,1))
ggsave("combined_plot.png",width=4,height=12)

```

```{r LIconcordance}
#Logic of analysis: L-lateralised twin with bilat or R-lat MZ cotwin
#more likely to lack right shift factor
#Predict lower language
#So focus here is on MZ L-lateralised only, subcategorised by cotwin lat.

concordMZ <- filter(doubledata,MZDZ==1,li_cat==1)
concordMZ$cotwinsame<-1
w<-which(concordMZ$li_cat2>1)
concordMZ$cotwinsame[w]<-0
t.test(concordMZ$n_language_low~concordMZ$cotwinsame) #no hint of any difference

```

```{r testnormality}
#--------------------------------------------------------------
#It's clear handedness all measures are v non-normal.
#--------------------------------------------------------------
donorm<-0
if (donorm==1){
shapiro.test(doubledata$ehp_handedness)
qqnorm(doubledata$ehp_handedness)
shapiro.test(doubledata$qhp_handedness)
qqnorm(doubledata$qhp_handedness)
shapiro.test(doubledata$lat_index)
qqnorm(doubledata$lat_index)
shapiro.test(doubledata$taskssum)
qqnorm(doubledata$taskssum)
}
```
```{r simhand, echo=FALSE}
#--------------------------------------------------------------
#Check impact of non-normality on heritability estimates
#If you have normal data but you censor scores so 75% are the same, how does this affect correlation?
#--------------------------------------------------------------
dosim <-0
if(dosim==1){
nsims <-100
mysimdat<-data.frame(matrix(NA,nrow=nsims,ncol=4))
colnames(mysimdat)<-c('Original_r','Original_rho','Censored_r','Censored_rho')
myn <- 100
truer<-0
mycov <-matrix(c(1,truer,truer,1),nrow=2)
for (i in 1:nsims){
simdat <- mvrnorm(myn , mu = c(0,0), mycov)
#now censor values, so that all > -.5 are 0
simdat2<-simdat
w<-which(simdat[,1]>-.5)
simdat2[w,1]<-0
w<-which(simdat[,2]>-.5)
simdat2[w,2]<-0
r1<-cor(simdat,method='pearson')
r2<-cor(simdat,method='spearman')
mysimdat[i,1]<-r1[1,2]
mysimdat[i,2]<-r2[1,2]

r3<-cor(simdat,method='pearson')
r4<-cor(simdat2,method='spearman')
mysimdat[i,3]<-r3[1,2]
mysimdat[i,4]<-r4[1,2]
}
round(colMeans(mysimdat),3)

#results show censoring has minimal effect - serves to reduce correlation if rho used.
}
```

```{r AEmodel, echo=FALSE}
#---------------------------------------------------------------
#Try fitting AE model, despite low correlations.
#--------------------------------------------------------------
# Predict it won't fit because of zero DZ correlation - only compatible with no genetic effect! 
# Surprisingly, fit seems OK.
## NB However, really need a model that takes into account nonnormality! - see below re ordinal data
nv <- 1 #N variables
ntv <- nv*2 #N columns (twins)
aatable <-data.frame(matrix(NA,nrow=3,ncol=6))
colnames(aatable)<-c('Measure','N MZ','N DZ','a','a.SE','a^2')
mxcols <-matrix(c('ehp_handedness','ehp_handedness2','qhp_handedness','qhp_handedness2','lat_index','lat_index2'),nrow=3,byrow=TRUE)
for (mytask in 1:3){
selVars <- mxcols[mytask,1:2]
latcols<-which(names(doubledata)%in%selVars)
rowrange<-1:(nrow(doubledata)/2)
mymz <- filter(doubledata[rowrange,],MZDZ==1)
mydz <- filter(doubledata[rowrange,],MZDZ==2)
if(mytask==3)
{mymz<-filter(mymz,doppout==0)
mydz<-filter(mydz,doppout==0)
}
MZdata  <-  mxData(scale(mymz[,latcols]), type="raw" )
DZdata  <-  mxData(scale(mydz[,latcols]), type="raw" )

a <-  mxMatrix( type="Lower", nrow=nv, ncol=nv, free=TRUE, values=.6, label="a11", name="a" )
e <-  mxMatrix( type="Lower", nrow=nv, ncol=nv, free=TRUE, values=.6, label="e11", name="e" )

A <-  mxAlgebra( expression=a %*% t(a), name="A" )
E <-  mxAlgebra( expression=e %*% t(e), name="E" )

Mean    <-  mxMatrix( type="Full", nrow=1, ncol=nv, free=TRUE, values= 0, label="mean", name="Mean" )
expMean <-  mxAlgebra( expression= cbind(Mean,Mean), name="expMean")

expCovMZ <- mxAlgebra( expression= rbind  ( cbind(A+E , A),cbind(A   , A+E)), name="expCovMZ" )
expCovDZ <- mxAlgebra( expression= rbind  ( cbind(A+E     , 0.5%x%A),cbind(0.5%x%A , A+E)),  name="expCovDZ" ) 

obs <-  list(a,e,A,E,Mean,expMean,expCovMZ,expCovDZ)

fun <- mxFitFunctionML()
mzExp <- mxExpectationNormal(covariance="expCovMZ", means="expMean", dimnames=selVars )
dzExp <- mxExpectationNormal(covariance="expCovDZ", means="expMean", dimnames=selVars )

MZ <- mxModel("MZ", obs, MZdata, fun, mzExp)
DZ <- mxModel("DZ", obs, DZdata, fun, dzExp)

aeFun <- mxFitFunctionMultigroup(c("MZ","DZ"))
ae <- mxModel("AE", MZ, DZ, fun, aeFun)

aeFit <- mxRun(ae, silent = T)
summary(aeFit)

pars <- round(cbind(aeFit$output$estimate, aeFit$output$standardErrors), 3)
colnames(pars) <- c("Estimates", "Std. Err.")
aatable[mytask,1]<-mxcols[mytask,1]
aatable[mytask,2]<-nrow(mymz)
aatable[mytask,3]<-nrow(mydz)
aatable[mytask,4]<-pars[1,1]
aatable[mytask,5]<-pars[1,2]
aatable[mytask,6]<-round(pars[1,1]^2,2)
}
stargazer(aatable,type='text',title='Table 3: Heritability estimates from AE model for different measures: ignoring nonnormality',summary=FALSE)

#NB all tasks have been scaled; skew in the handedness measures has not been accounted for
```

```{r polychoric, echo=FALSE}
#---------------------------------------------------------------
dopoly<-1
if(dopoly==1){
#AE model; try ordinal version for handedness.
# Based on ibg.colorado.edu/cdrom2010/medland/wed_morning/wed_morning2010.pdf


#use double-entry file ? otherwise problems if null value for one category
myehp<-dplyr::select(doubledata,MZDZ,ehp_handedness,ehp_handedness2)
npair<-nrow(myehp)
my.mz<-filter(myehp[1:npair,],MZDZ==1)
mypolytab.mz<-table(my.mz$ehp_handedness,my.mz$ehp_handedness2)
my.dz<-filter(myehp[1:npair,],MZDZ==2)
mypolytab.dz<-table(my.dz$ehp_handedness,my.dz$ehp_handedness2)
mypoly.mz<-polychoric(mypolytab.mz)$rho
mypoly.dz<-polychoric(mypolytab.dz)$rho
print('Polychoric correlations MZ/DZ (rho)')
print(c(mypoly.mz,mypoly.dz))


myqhp<-dplyr::select(doubledata,MZDZ,ehp_handedness,ehp_handedness2)
npair<-nrow(myehp)
my.mz<-filter(myehp[1:npair,],MZDZ==1)
mypolytab.mz<-table(my.mz$ehp_handedness,my.mz$ehp_handedness2)
my.dz<-filter(myehp[1:npair,],MZDZ==2)
mypolytab.dz<-table(my.dz$ehp_handedness,my.dz$ehp_handedness2)
mypoly.mz<-polychoric(mypolytab.mz)$rho
mypoly.dz<-polychoric(mypolytab.dz)$rho

#Is result the same as for ranked values?
my.mz$temp<-rank(my.mz$ehp_handedness)
my.mz$temp2<-rank(my.mz$ehp_handedness2)
cor(my.mz$temp,my.mz$temp2,method='spearman')
#No! polychoric rho is higher - assumes biv normality?

#Is result the same if values assigned as z-s by freq
#Need z corresponding to cumulative freq for each score
#first use table function to get freq distribution
myhist <-table(data.short$ehp_handedness)
myhist <-myhist/sum(myhist) #convert to proportions
#make into cumulative freq dist
myhist2<-vector()
myhist2[1]<-myhist[1]
for (i in 2:length(myhist)){
  myhist2[i]<-myhist[i]+myhist2[i-1]
}

mythresh<-vector()
mythresh[1]<- -2 #floor value
for (i in 1:(length(myhist)-1))
{mythresh[i+1]<-qnorm(myhist2[i])
  
}
tempvec<-my.mz$ehp_handedness+1 #Cannot have index of zero: need to add one to score so it can act as index
my.mz$zhand<-mythresh[tempvec]
tempvec2<-my.mz$ehp_handedness2+1
my.mz$zhand2<-mythresh[tempvec2]
cor(my.mz$zhand,my.mz$zhand2,method="spearman")
#Still much lower than value from polychoric!
}
```
```{R umxordinal, echo=TRUE}
# ===================
# = Ordinal example from Bates with handedness data substituted=
# ===================
require(umx)
doubledata$ehp_handedness1<-as.factor(doubledata$ehp_handedness) #just to fit labelling convention
doubledata$ehp_handedness2<-as.factor(doubledata$ehp_handedness2)

ordDVs = c("ehp_handedness1", "ehp_handedness2")
selDVs = c("ehp_handedness")
# Make the ordinal variables into mxFactors (ensure ordered is TRUE, and require levels)
singledata<-doubledata[doubledata$twin==1,]
singledata[, ordDVs] <- mxFactor(singledata[, ordDVs], levels = c(0:10))

mzData <- singledata[singledata$MZDZ ==1, ]
dzData <- singledata[singledata$MZDZ ==2, ]


m1 = umxACE(selDVs = selDVs, dzData = dzData, mzData = mzData, sep = '',thresholds='deviationBased')
umxSummary(m1)
parameters(m1)

```

##Thoughts
Other studies find high heritability for raw L and R measures - it's just the asymmetry score that is not high.
I found that for hand skill - also for brain things
Would we find the same here? If so, what would it mean?
Is it raw L hem power that matters? Or raw combined power?
So I need to go back and get measure of flow in POI

```{r poiavgs, echo=FALSE}
#--------------------------------------------------------------
# Analysis of poi averages
#--------------------------------------------------------------
dopoi<-0
if (dopoi==1){
  datadir <-"~/Doppler_analysis/"
rawfile<-"longrawDopplermeans.csv" #holds all normalised values by subject/side/timepoint
rawdata<-read.csv(paste0(datadir,rawfile))
cutdata<-filter(rawdata,!is.na(value)) #remove those with missing data
cutdata<-filter(cutdata,time>3.99&time<14.01) #focus only on POI
cutdata<-cutdata[,c(1,2,4)] #don't need time variable; averaging over this
aggdata <-aggregate(cutdata$value, by=list(cutdata$ID,cutdata$side),
  FUN=mean)
colnames(aggdata)<-c('ID','side','flow')
#do next bit in a loop, though should be possible to avoid this
colstart<-ncol(doubledata2)+1
doubledata2[,colstart:(colstart+3)]<-NA
colnames(doubledata2)[colstart:(colstart+3)]<-c('MeanL1','MeanR1','MeanL2','MeanR2')
for (i in 1:(nrow(doubledata2)/2)){
  myid<-doubledata2$record_id[i]
  w<-which(aggdata$ID==myid)
  if(length(w)>0){  #some IDs are unexpectedly missing?
  doubledata2[i,colstart:(colstart+1)]<-aggdata$flow[w]
   myid2<-doubledata2$record_id2[i]
  w<-which(aggdata$ID==myid2)
  doubledata2[i,(colstart+2):(colstart+3)]<-aggdata$flow[w] 
  } 
}
#Need to check why some datapoints are omitted from the longxls file.
#This is just a kludge to get idea of if this is going to be worth pursuing
#In general, no. Confirms that the absdiff scores v highly correlated with LI.
#But no evidence for MZ>DZ correlations for the abs values or the diff
doubledata2[141:280,42:43]<-doubledata2[1:140,44:45]
doubledata2[141:280,44:45]<-doubledata2[1:140,42:43]
mymz<-which(doubledata2$MZDZ==1)
mydz<-which(doubledata2$MZDZ==2)

doubledata2$absdiff1<-doubledata2$MeanL1-doubledata2$MeanR1
doubledata2$absdiff2<-doubledata2$MeanL2-doubledata2$MeanR2

doubledata2$propL1<-doubledata2$MeanL1/(doubledata2$MeanL1+doubledata2$MeanR1)
doubledata2$propL2<-doubledata2$MeanL2/(doubledata2$MeanL2+doubledata2$MeanR2)
doubledata2$totflow<-doubledata2$MeanL1+doubledata2$MeanR1
doubledata2$totflow2<-doubledata2$MeanL2+doubledata2$MeanR2
corm<-round(cor(doubledata2[mymz,c(3,6,10,30,44:53)],use='complete.obs'),3)
cord<-round(cor(doubledata2[mydz,c(3,6,10,30,44:53)],use='complete.obs'),3)

}
```

```{r medland, echo=FALSE}
#--------------------------------------------------------------
# Look at Medland handedness data
#--------------------------------------------------------------
domedland=0
if(domedland==1){#Issues around Medland meta-analysis of handedness- when a2 is v low, can get -ve estimates.
#If these are censored at zero, will bias overall picture.
#Qu of what things look like if they are included.
#Also - if DZ correl is NS, any genetic influence seems implausible?
medland<-read.csv('medlandcsv.csv')
for (i in 1:nrow(medland)){
 vec<-c(medland$MZ_RR[i],medland$MZ_mixed[i]/2,medland$MZ_mixed[i]/2,medland$MZ_LL[i]) 
 medland$phi.m[i]<-phi(vec,digits=3)
  vec2<-c(medland$DZ_RR[i],medland$DZ_mixed[i]/2,medland$DZ_mixed[i]/2,medland$DZ_LL[i]) 
 medland$phi.d[i]<-phi(vec2,digits=3)

}
#Check phi coeffs for mz and dz: note dz are v low indeed
par(mfrow=c(1,2))
 plot(log(medland$totN),medland$phi.d,col=(1+medland$measuretype),main='DZ')
 abline(h=0)
 # Add a legend
legend(7, .5, legend=c('dk','writing','other uni','multiple','skill','self-report'),
       col=seq(1:7), lty=1, cex=0.8)

 plot(log(medland$totN),medland$phi.m,col=(1+medland$measuretype),main='MZ')
 abline(h=0)
 # Add a legend
legend(7, .6, legend=c('dk','writing','other uni','multiple','skill','self-report'),
       col=seq(1:7), lty=1, cex=0.8)

#Estimate heritability from phi coeffiences (twice diffc between MZ and DZ)
 medland$phi_heritab<-2*(medland$phi.m-medland$phi.d)
 hist(medland$phi_heritab)
 #How does heritability depend on sample size
 medland$totN<-medland$MZ_total+medland$DZ_total
 plot(log(medland$totN),medland$phi_heritab,col=(1+medland$measuretype))
 # Add a legend
legend(6, -.5, legend=c('dk','writing','other uni','multiple','skill','self-report'),
       col=seq(1:7), lty=1, cex=0.8)

#focus just on those with big N
 medlandshort<-filter(medland,totN>200)
 plot(medlandshort$year,medlandshort$phi_heritab)
 hist(medland$phi.d)
}
```
##Session information
```{r sessinfo}
sessionInfo()
```
##References

Annett, M. (1985). Left, right, hand and brain: the Right Shift Theory. Hillsdale, NJ: Erlbaum.

McManus , I. C. (1985). Handedness, language dominance and aphasia: a genetic model. Psychological Medicine, Monograph Supplement, 8. 

Medland, S. E., Duffy, D. L., J.Wright, M., Geffen, G. M., Hay, D. A., Levy, F., . . . Boomsma, D. I. (2009). Genetic influences on handedness: Data from 25,732 Australian and Dutch twin families. Neuropsychologia, 47, 330-337. 

Medland, S. E., Duffy, D. L., Wright, M. J., Geffen, G. M., & Martin, N. G. (2006). Handedness in twins: Joint analysis of data from 35 samples. Twin Research and Human Genetics, 9, 46-53. 

R Core Team (2013). R: A language and environment for statistical computing. R Foundation for Statistical Computing, Vienna, Austria

Turkheimer E. (2000). Three laws of behavior genetics and what they mean. Current Directions in Psychological Science, 9, 160–164. 

Tenesa A., Haley C. S. (2013). The heritability of human disease: Estimation, uses and abuses. Nature Reviews Genetics, 14, 139–149.